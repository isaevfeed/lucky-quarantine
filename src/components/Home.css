.Home_current-day {
    display: flex;
    position: absolute;
    flex-direction: column;
    align-items: center;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}
.Home_current-day-img {
    width: 175px;
    margin-bottom: 20px;
}
.Home_current-day-title {
    font-size: 24px;
    font-weight: normal;
    padding: 0;
    margin: 0;
}