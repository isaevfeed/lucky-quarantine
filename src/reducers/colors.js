const colors = {
    'food': '#BDFBFF',
    'break': '#C2FFBD',
    'sleep': '#E7E7E7',
    'task': '#FEFFBD',
    'lesson': '#FEBDFF',
    'job' : '#FFBDBD',
};

export default function(state = colors, payload) {
    switch(payload.type) {
        default: 
            return state;
    }
}