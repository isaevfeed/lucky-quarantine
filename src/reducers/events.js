const events = [
    {
        type: 'sleep',
        subtitle: 'Wake up',
        content: null,
        time: '8:00',
    },
    {
        type: 'food',
        subtitle: 'Break',
        content: null,
        time: '21:05',
    },
];

export default function(state = events, payload) {
    switch(payload.type) {
        default:
            return state
    }
}