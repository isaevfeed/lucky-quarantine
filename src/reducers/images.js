const images = {
    'food': 'burger',
    'break': 'clock',
    'sleep': 'clock',
    'task': 'puzzle-pieces',
    'lesson': 'contract',
    'job' : 'salary',
};

export default function(state = images, action) {
    switch(action.type) {
        default:
            return state;
    }
}