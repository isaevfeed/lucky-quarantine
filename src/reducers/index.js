import {combineReducers} from 'redux';
import eventReducer from './events';
import taskReducer from './tasks';
import colorReducer from './colors';
import imageReducer from './images';

const reducers = combineReducers({
    evarr: eventReducer,
    colors: colorReducer,
    images: imageReducer,
    tasks: taskReducer,
});

export default reducers;