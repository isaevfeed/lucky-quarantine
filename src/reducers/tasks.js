const tasks = [];

export default function(state = tasks, payload) {
    if (localStorage && localStorage.getItem('tasks')) {
        const localTasks = localStorage.getItem('tasks');
        if (localTasks != '' && localTasks != '[]') {
            state = JSON.parse(localTasks);
        }
    }
    console.log(state);
    switch(payload.type) {
        case 'ADD':
            state = [...state, payload.task];
            saveTaskInStorage(state)
            return state;
        case 'COMPLETE':
            state = state.map(task => {
                if (task.id == payload.id) {
                    task.status = !task.status;
                }
                return task;
            });
            saveTaskInStorage(state)
            return state;
        default:
            saveTaskInStorage(state)
            return state;
    }
}

function saveTaskInStorage(state) {
    localStorage.setItem('tasks', JSON.stringify(state));
}