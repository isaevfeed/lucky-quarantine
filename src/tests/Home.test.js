import ReactDOM from 'react-dom';
import Home from '../components/App/Home';
import {images} from '../reducers/images';
import renderer from 'react-test-renderer';

const testEventsData = [
    {
        type: 'sleep',
        subtitle: 'Wake up',
        content: null,
        time: '8:00',
    },
];

const TestEventComponent = () => <Home items={testEventsData} images={images} />;

it('Home (render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TestEventComponent />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('Home Snapshot', () => {
    const snap = renderer.create(<TestEventComponent />).toJSON();
    expect(snap).toMatchSnapshot();
});