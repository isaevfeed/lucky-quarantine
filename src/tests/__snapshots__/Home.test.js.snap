// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot 1`] = `
<div
  className="Home_current-day"
>
  <img
    className="Home_current-day-img"
    src="/images/clock.png"
  />
  <h4
    className="Home_current-day-title"
  >
    Wake up
  </h4>
</div>
`;
